kyaml/pathutil/pathutil.go

```diff
@@ -9,20 +9,27 @@ import (
 	"strings"
 )
 
-// SubDirsWithFile takes the root directory path and returns all the paths of
-// sub-directories which contain file with input fileName including itself
-func SubDirsWithFile(root, fileName string) ([]string, error) {
+// DirsWithFile takes the root directory path and returns all the paths of
+// sub-directories(including itself) which contain file with input fileName
+// at top level if recurse is true
+func DirsWithFile(root, fileName string, recurse bool) ([]string, error) {
 	var res []string
+	if !recurse {
+		// check if the file with fileName is present in root and return it
+		// else return empty list
+		_, err := os.Stat(filepath.Join(root, fileName))
+		if !os.IsNotExist(err) {
+			res = append(res, root)
+		}
+		return res, nil
+	}
 	err := filepath.Walk(root,
-		func(path string, info os.FileInfo, err error) error {
+		func(filePath string, info os.FileInfo, err error) error {
```

Comment:
Use a different name for the first parameter here. Having a variable named `filePath` and an imported package called `filepath` is confusing.

Reply:
Done.